src/utils/dom.js

```diff
@@ -11,10 +11,10 @@
  * @returns {HTMLElement} Parent with specified attribute value.
  */
 export function parentWithAttribute(elem, name, value) {
-    while ((value ? elem.getAttribute(name) !== value : !elem.getAttribute(name))) {
+    while ((value ? elem.dataset.name !== value : !elem.dataset.name)) {
```

Comment:
Missing brackets - `[name]`. But what if the attribute is not from the `dataset`?